src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -222,18 +222,26 @@ define([
       }
     },
 
-    destroyDeployment: function(deployment, component) {
+    destroyDeployment: function(deployment, component, force) {
       component.setLoading(true);
-      if (confirm("Destroy deployment of apps: '" + deployment.affectedAppsString() +
-          "'?\nDestroying this deployment will create and start a new deployment to revert the affected app to its previous version.")) {
 
+      var confirmMessage = !force ?
+        "Destroy deployment of apps: '" + deployment.affectedAppsString() +
+          "'?\nDestroying this deployment will create and start a new deployment to revert the affected app to its previous version." :
+        "Stop deployment of apps: '" + deployment.affectedAppsString() +
```

Comment:
The async 'HTTP 202 Accepted' must be handled correctly by the error callback, because it's not an error. 
Deployment shouldn't show up anymore after pressing the button. There was also an error in setLoading(false). So I removed it.
Don't have a clue why you get "Stop deployment of apps: ''?, that is working fine here.
This function is not working.
```
affectedAppsString: function() {
  return this.get("currentActions").affectedApps.join(", ");
}
```